### PR TITLE
Customer requested changes for AccuRev Plugin.

### PR DESCRIPTION
**Pull request by PoojaVishal**
- AccuRev Jenkins Plug-in: Misspelling and capitalization of product name AccuRev.

- AccuRev Jenkins Plug-in repeated message console output: "No Accurv tool is chosen, reverting to Default" is excess noise.

- Jenkins job should be FAILURE if AccuRev command(checkout/populate/changelogset) fails.

### CODE REVIEW

**Comment by PoojaVishal** [comment]
Hi Joseph,

Please review my changes.

**Comment by casz** [comment]
@PoojaVishal hi could you please squash your commits 😊

**Comment by PoojaVishal** [comment]
check now..

**Comment by PoojaVishal** [comment]
@casz  how to add you in reviewers?
and how to check if i have squashed my commits?

**Comment by casz** [comment]
@PoojaVishal when you only have one commit here https://github.com/jenkinsci/accurev-plugin/pull/63/commits

**Comment by PoojaVishal** [comment]
done.

**Comment by PoojaVishal** [comment]
@casz I made one more small change in config.jelly. if i push that will it be included in this push?

**Comment by casz** [comment]
@PoojaVishal yes it would 👍 anything commits added to your branch

**Comment by PoojaVishal** [comment]
format as in? which class?

File: src/main/java/hudson/plugins/accurev/AccurevLauncher.java [hunk 1]
```diff
@@ -291,31 +291,32 @@ public static boolean runCommand(//
...
-                    "Selected Accurev installation does not exist. Using Default %s (%s)",
-                    accurev.getName(),
-                    accurev.getHome()
-                )
-            );
+    private static void handleConsoleMessage(TaskListener listener, boolean isValidCommand,
+            String path) {
+        boolean isValidPath = false;
+        isValidPath = path!=null && (path.isEmpty() || path.equals("accurev"));
+        if(isValidCommand && isValidPath)
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevLauncher.java]: This has several formatting issues

File: src/main/java/hudson/plugins/accurev/delegates/AbstractModeDelegate.java [hunk 2]
```diff
@@ -180,7 +180,28 @@ public boolean checkout(Run<?, ?> build, Launcher launcher, FilePath jenkinsWork
             setStreamColor();
         }
 
-        return checkout(build, changelogFile) && populate(build) && captureChangeLog(build, changelogFile, streams);
+        boolean isSuccess = true;
+        String errorMessage = "Failed to";
+        if(!checkout(build, changelogFile) ){
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/delegates/AbstractModeDelegate.java]: This is also not formatted correctly

**Comment by PoojaVishal** [comment]
Please check now.

File: src/main/java/hudson/plugins/accurev/AccurevLauncher.java [hunk 3]
```diff
@@ -291,31 +291,26 @@ public static boolean runCommand(//
...
-                    "Selected Accurev installation does not exist. Using Default %s (%s)",
-                    accurev.getName(),
-                    accurev.getHome()
-                )
-            );
+    private static void handleConsoleMessage(TaskListener listener, boolean isValidCommand, String path) {
+        boolean isValidPath = false;
+        isValidPath = (path != null && (path.isEmpty() || path.equals("accurev")));
+        if (isValidCommand && isValidPath) {
+            listener.getLogger().println(String.format("No AccuRev tool configured, using default"));
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevLauncher.java]: Think we can skip the String.format here

**Comment by PoojaVishal** [comment]
Check now..
I hope will not ask me to squash comments again..

File: src/main/java/hudson/plugins/accurev/AccurevLauncher.java [hunk 4]
```diff
@@ -291,31 +291,26 @@ public static boolean runCommand(//
...
-            );
-            return defaultInstallation;
+    public static AccurevTool resolveAccurevTool(String accurevToolValue, TaskListener listener, String command) {
+        AccurevTool accurevTool = Jenkins.getInstance().getDescriptorByType(AccurevTool.DescriptorImpl.class).getInstallation(accurevToolValue);
+        boolean isValidCommand = command.equals("info")||command.equals("login");
+        if (accurevTool == null ) {
+            accurevTool = AccurevTool.getDefaultInstallation();
+            String path = accurevTool.getHome();
+            handleConsoleMessage(listener, isValidCommand, path);
+        }else{
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevLauncher.java]: spacing is off

File: src/main/java/hudson/plugins/accurev/AccurevLauncher.java [hunk 5]
```diff
@@ -291,31 +291,26 @@ public static boolean runCommand(//
...
-                    "No Accurv tool is chosen, reverting to %s (%s)",
-                    defaultInstallation.getName(),
-                    defaultInstallation.getHome()
-                )
-            );
-            return defaultInstallation;
+    public static AccurevTool resolveAccurevTool(String accurevToolValue, TaskListener listener, String command) {
+        AccurevTool accurevTool = Jenkins.getInstance().getDescriptorByType(AccurevTool.DescriptorImpl.class).getInstallation(accurevToolValue);
+        boolean isValidCommand = command.equals("info")||command.equals("login");
+        if (accurevTool == null ) {
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevLauncher.java]: spacing is off

File: src/main/java/hudson/plugins/accurev/AccurevLauncher.java [hunk 6]
```diff
@@ -291,31 +291,26 @@ public static boolean runCommand(//
...
-                String.format(
-                    "No Accurv tool is chosen, reverting to %s (%s)",
-                    defaultInstallation.getName(),
-                    defaultInstallation.getHome()
-                )
-            );
-            return defaultInstallation;
+    public static AccurevTool resolveAccurevTool(String accurevToolValue, TaskListener listener, String command) {
+        AccurevTool accurevTool = Jenkins.getInstance().getDescriptorByType(AccurevTool.DescriptorImpl.class).getInstallation(accurevToolValue);
+        boolean isValidCommand = command.equals("info")||command.equals("login");
```
Comment (casz) [inline src/main/java/hudson/plugins/accurev/AccurevLauncher.java]: spacing is off

**Comment by PoojaVishal** [comment]
@casz  can you generate a public build for final release?
this build has snapshot in its version number.

**Comment by casz** [comment]
I'll spin a release tonight 😎

**Comment by PoojaVishal** [comment]
@casz  please let me know once you trigger a release build.

**Comment by casz** [comment]
@PoojaVishal done 😅

**Comment by PoojaVishal** [comment]
@casz  the invitation you sent me for collaborator i accepted it, will that give me permission to build and publish? 
or should i follow the steps you sent to sign up in Jenkins and login etc..

**Pull request merged**